sqlmodel/main.py

```diff
@@ -610,16 +604,7 @@ def _calculate_keys(  # type: ignore
             return self.__fields__.keys()  # | self.__sqlmodel_relationships__.keys()
 
         keys: AbstractSet[str]
-        if exclude_unset:
-            keys = self.__fields_set__.copy()
-        else:
-            # Original in Pydantic:
-            # keys = self.__dict__.keys()
-            # Updated to not return SQLAlchemy attributes
-            # Do not include relationships as that would easily lead to infinite
-            # recursion, or traversing the whole database
-            keys = self.__fields__.keys()  # | self.__sqlmodel_relationships__.keys()
-
+        keys = self.__fields_set__.copy() if exclude_unset else self.__fields__.keys()
```

Comment:
I prefer to keep the previous version with the comments explaining what's happening.